ompi/communicator/comm_init.c

```diff
@@ -300,6 +302,16 @@ int ompi_comm_init_mpi3 (void)
        MPI_COMM_SELF, the keyhash will automatically be created. */
     ompi_mpi_comm_self.comm.c_keyhash = NULL;
 
+    char *memkind_requested = getenv ("OMPI_MCA_mpi_memory_alloc_kinds");
```

Comment:
This should probably be an MCA parameter so it can be set via `--mca mpi_memory_alloc_kinds`. Maybe something along the lines of:

```C
const char *memkind_requested = NULL;
mca_base_var_register("mpi", "", "memory", "alloc_kinds",
                      "Requested MPI memory allocation kinds",
                      MCA_BASE_VAR_TYPE_STRING, NULL, 0,
                      MCA_BASE_VAR_FLAG_DEFAULT_ONLY, OPAL_INFO_LVL_1,
                      MCA_BASE_VAR_SCOPE_CONSTANT, &memkind_requested);
```

I'm not sure this is entirely correct, maybe @bosilca can chime in.

Reply:
This is alreadyt an MCA parameter, its defined in the PRRTE layer, and is used to implement the `-memory-alloc-kinds` argument of the mpiexec command line.